### PR TITLE
fix: Log error messages if there are errors converting statistics in read buffer

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
I am tracking down https://github.com/influxdata/influxdb_iox/issues/2217 and I don't know what the underlying issue is yet. I have a theory that it is a panic within `ReadBuffer::Chunk::table_summary` (which is consistent with the panic seen in #2217) and so this PR is designed to let me prove/disprove that theory

Note is code that was recently changed in https://github.com/influxdata/influxdb_iox/pull/2159

Changes:
1. Add fallible conversion of metadata -> `TableSummary` in the read buffer
2. Log the error if there is an error prior to panic'ing

### CODE REVIEW

File: read_buffer/src/table.rs [hunk 1]
```diff
@@ -686,6 +738,19 @@ impl MetaData {
     }
 
     pub fn to_summary(&self, table_name: impl Into<String>) -> TableSummary {
+        match self.to_summary_fallable(table_name) {
+            Ok(summary) => summary,
+            Err(e) => {
+                error!(%e, "Unexpected error converting metadata to TableSummary");
```
Comment (alamb) [inline read_buffer/src/table.rs:744]: The whole point of this PR is this log statement

**Review by tustvold — APPROVED** [review]
Tentatively approved, but I'm not really sure what additional benefit the error plumbing is providing - it seems to just move the panic and make the stack trace less informative?

**Comment by alamb** [comment]
> Tentatively approved, but I'm not really sure what additional benefit the error plumbing is providing - it seems to just move the panic and make the stack trace less informative?

@tustvold  the core problem is that the `panic` message from the stack trace is not in the logs for some reason. Thus the error plumbing acrobatics are to try and capture what is erroring, if indeed this is even the code path that is doing so. 


```
stack backtrace:
   0: rust_begin_unwind
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:515:5
   1: std::panicking::begin_panic_fmt
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/std/src/panicking.rs:457:5
   3: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &mut F>::call_once
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/ops/function.rs:280:13
   4: core::option::Option<T>::map
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/option.rs:489:29
   5: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::next
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/iter/adapters/map.rs:101:9
   6: alloc::vec::Vec<T,A>::extend_desugared
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/alloc/src/vec/mod.rs:2546:35
   7: <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<T,I>>::spec_extend
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/alloc/src/vec/spec_extend.rs:18:9
   8: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/alloc/src/vec/spec_from_iter_nested.rs:37:9
   9: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/alloc/src/vec/spec_from_iter.rs:33:9
  10: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/alloc/src/vec/mod.rs:2449:9
  11: core::iter::traits::iterator::Iterator::collect
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/iter/traits/iterator.rs:1748:9
  15: server::db::catalog::chunk::CatalogChunk::new_rub_chunk
             at ./home/rust/project/server/src/db/catalog/chunk.rs:306:41
  16: server::db::catalog::partition::Partition::create_rub_chunk
             at ./home/rust/project/server/src/db/catalog/partition.rs:170:58
  17: server::db::lifecycle::compact::compact_chunks::{{closure}}
             at ./home/rust/project/server/src/db/lifecycle/compact.rs:103:13
  18: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at ./rustc/a178d0322ce20e33eac124758e837cbd80a6f633/library/core/src/future/mod.rs:80:19
...

```

**Comment by alamb** [comment]
Turns out the lack of message in the panic was user error (I had too many greps). https://github.com/influxdata/influxdb_iox/issues/2240#issuecomment-895966935 

This PR is not needed. 

(for the record I was doing 
```
kubectl -n iox logs svc/iox-query-3000 -c iox   | grep -v read_buffer | grep -v "Chunk to persist has outstanding action" | grep -v "persisting partition as exceeds row threshold" > ~/Downloads/panic.log2
```
)

aaand the panic message has `read_buffer` 🤦

**Pull request closed without merging**